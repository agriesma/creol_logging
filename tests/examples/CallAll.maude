load creol-interpreter .
mod PROGRAM is
  protecting CREOL-SIMULATOR .
  op classes : -> Configuration [ctor] .
  eq classes =
    < "CallAll" : Class | Inh: noInh, Param: noVid, Att: "outstanding" |->
      null, "list" |-> null, "results" |-> null,
      Mtds: < "init" : Method | Param: noVid, Att: "_" |-> null, Code: skip ;
              return ( emp ) >,
            < "run" : Method | Param: noVid, Att: "_" |-> null, Code: skip ;
              return ( emp ) >,
            < "doit" : Method | Param: "l", Att: "l" |-> null, "label:0" |->
              null, "i" |-> null, "r" |-> null, "o" |-> null, "_" |-> null,
              Code: if "/=" ( "l" :: list(emp) ) th call( "r" ;
              "fst" ( "i" ) ; "m" ; "snd" ( "i" ) ) ;
              call( "label:0" ; "this" ; "doit" ; "tail" ( "l" ) ) ;
              get( "label:0" ; noVid ) ; free( "label:0" ) ; await ?("r") ;
              get( "r" ; "o" ) ; free( "r" ) ;
              assign( "results" ; "|-" ( "results" :: list("fst" ( "i" ) ::
              "o") ) ) el skip fi ; return ( emp ) >,
            < "call_all" : Method | Param: noVid, Att: "label:0" |-> null,
              "_" |-> null, Code: assign( "results" ; list(emp) ) ;
              call( "label:0" ; "this" ; "doit" ; "list" ) ;
              get( "label:0" ; noVid ) ; free( "label:0" ) ; await
              "=" ( "#" ( "results" ) :: "#" ( "list" ) ) ; return ( emp ) >,
      Ocnt: 0 > .
endm

