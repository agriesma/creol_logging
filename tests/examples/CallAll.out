==========================================
reduce in PROGRAM : classes .
result Class: 
< "CallAll" : Class | Inh: noInh, Param: noVid, Att: "list" |-> null,
    "outstanding" |-> null, "results" |-> null, Mtds:
  < "call_all" : Method | Param: noVid, Att: "_" |-> null, "label:0" |-> null,
    Code: assign("results" ; list(emp)) ; call("label:0" ; "this" ; "doit" ;
    "list") ; get("label:0" ; noVid) ; free("label:0") ; await "="("#"(
    "results") :: "#"("list")) ; return(emp) >, < "doit" : Method | Param: "l",
    Att: "_" |-> null, "i" |-> null, "l" |-> null, "label:0" |-> null, "o" |->
    null, "r" |-> null, Code: if "/="("l" :: list(emp)) th call("r" ; "fst"(
    "i") ; "m" ; "snd"("i")) ; call("label:0" ; "this" ; "doit" ; "tail"("l"))
    ; get("label:0" ; noVid) ; free("label:0") ; await ?("r") ; get("r" ; "o")
    ; free("r") ; assign("results" ; "|-"("results" :: list("fst"("i") ::
    "o"))) el skip fi ; return(emp) >, < "init" : Method | Param: noVid, Att:
    "_" |-> null, Code: skip ; return(emp) >, < "run" : Method | Param: noVid,
    Att: "_" |-> null, Code: skip ; return(emp) >,
  Ocnt: 0 >

Bye.
